lib/_stream_writable.js

```diff
@@ -108,6 +108,33 @@ function WritableState(options, stream) {
 
   // True if the error was already emitted and should not be thrown again
   this.errorEmitted = false;
+
+  // count buffered requests
+  this.bufferedRequestCount = 0;
+
+  this.corkedRequestsFree = new CorkedRequest(this);
+  this.corkedRequestsFree.next = new CorkedRequest(this);
+}
+
+function CorkedRequest(state) {
```

Comment:
Also, this is a linked list, is it not? It might benefit from a new name, as it isn't a single request.